### ISSUE TITLE
[Wayland] "Move window left/right outside management mode" doesn't handle multi-display setup correctly

### ISSUE THREAD

**Issue by evanmohr** [pop-os/shell#1575]
**(1) Issue/Bug Description:**
Using the "Move window left/right outside management mode" keyboard shortcut to swap positions of windows get result in broken (at at least unexpected if you are an i3/Sway user). Using this command can cause one window to be tiled to one direction but the other window will be tiled fullscreen under that.

![Diagram of bug](https://user-images.githubusercontent.com/80165419/214945477-65187497-b0fd-4d2d-af94-47c7e8649ad4.jpg)

**(2) Steps to reproduce (if you know):**

1. Tile two windows next to each other, I tested this using Firefox and Settings. Have window B as the active window.
![Example](https://user-images.githubusercontent.com/80165419/214945690-e27900f1-eecc-4ec7-a122-4a0b6e5e2595.jpg)
2. Hit the keyboard shortcut for "Move window left outside management mode". Note this this button is unbound by default. 
![This will (correctly) result window B moving to the left side of the screen](https://user-images.githubusercontent.com/80165419/214946001-22de0997-15fd-4d2b-8727-226065991474.jpg)
3. Hit the keyboard shortcut for "Move window left outside management mode". 
![This will (incorrectly) result in window B staying tiled to the left but window A becoming fullscreen under window B.](https://user-images.githubusercontent.com/80165419/214946167-c2cf8bfa-66ef-4abc-ba32-a78a0b00860c.jpg)

**(3) Expected behavior:**

As in i3 and Sway, I would expect for step 3 to move window B to my left monitor

**(4) Distribution (run `cat /etc/os-release`):**
 
NAME="Pop!_OS"
VERSION="22.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 22.04 LTS"
VERSION_ID="22.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=jammy
UBUNTU_CODENAME=jammy
LOGO=distributor-logo-pop-os

**(5) Gnome Shell version:**

42.3

**(6) Pop Shell version (run `apt policy pop-shell` or provide the latest commit if building locally):**
Use triple back ticks ```pop-shell:
  Installed: 1.1.0~1674603729~22.04~feaf0e3
  Candidate: 1.1.0~1674603729~22.04~feaf0e3
  Version table:
 *** 1.1.0~1674603729~22.04~feaf0e3 1001
       1001 http://apt.pop-os.org/release jammy/main amd64 Packages
       1001 http://apt.pop-os.org/release jammy/main i386 Packages
        100 /var/lib/dpkg/status```

**(7) Where was Pop Shell installed from:**

Clean install, just from apt.

**(8) Monitor Setup (2 x 1080p, 4K, Primary(Horizontal), Secondary(Vertical), etc):**

2x 2560x1440 monitors. The diagrams above occurred on my primary monitor. To the left, I have a second monitor.

**(9) Other Installed/Enabled Extensions:**

None

**(10) Other Notes:**

**Comment by jacobgkau** [comment]
I can't replicate with 1x 1080p and 1x 1440p (I don't have 2x 1440p with me right now), but from your description, it sounds like the issue is just that Window B isn't moving to the second display after attempting to move left from the left edge, correct? (Window A expands as if Window B is no longer on the display, but Window B is still on the display.)

**Comment by evanmohr** [comment]
I forgot to include this, but I am using the Wayland session.

**Comment by leftbones** [comment]
> I can't replicate with 1x 1080p and 1x 1440p (I don't have 2x 1440p with me right now), but from your description, it sounds like the issue is just that Window B isn't moving to the second display after attempting to move left from the left edge, correct? (Window A expands as if Window B is no longer on the display, but Window B is still on the display.)

Yes, this is exactly what happens. I am able to replicate this issue on a Wayland session with Pop Shell installed on Ubuntu 23.10

The shortcuts work normally, for example I can use left and right to swap two horizontally tiled windows around, but if I try to move the leftmost window left (towards my second monitor), then the above issue happens.

If I try to move a window in a direction where there is no other monitor (moving the rightmost window right) nothing happens, as expected. And the only way to clear up this weird issue is to toggle tiling on and off.

Edit: I also just found out that if I am on my second monitor (portrait) and have two windows tiled vertically, I can swap them normally with the up/down global move tile shortcuts. However, if I move the topmost window up, it moves to the right, onto my primary monitor? Very weird.
